### PR TITLE
Ge/fix component statement "part" form. Fixes #1232.

### LINKED ISSUE #1232: System Component Page missing part field, displaying remarks field incorrectly. Saving statement does not work.
**Describe the bug as a Story**
An information system's component page in the controls statements tab is missing the "part" field and incorrectly displaying the "remarks" field. Saving statement does not work.

Screen Shot 2020-12-28 at 4 30 26 PM

**To Reproduce**
1. Visit a system with controls/components set up
2. Click on 'Components'
3. Select the Component Statements tab
4. Click/open any component statement
5. Editing and saving a statement causes an error.

**Expected behavior**
Form should display the "part" field. Form should correctly render the "remarks" field underneath the "remarks" label. The form should save items.

**Screenshots**

Screen Shot 2020-12-28 at 4 30 26 PM

**Desktop (please complete the following information):**
 - Virtual Machine?
 - Screen size

## Developer hints

Possible issues is in template form

## Acceptance criteria 

  GIVEN this when I visit a system component
  WHEN I do open a component statement to edit
  THEN I can enter/edit a part. Update remarks. And save information.

### PR DESCRIPTION
**Pull request by gregelin**
This PR fixes the issue #1232 information system's component page in the controls statements tab is missing the "part" field and incorrectly displaying the "remarks" field. Saving statement does not work.

### CODE REVIEW

**Comment by gregelin** [comment]
Note, all controls passed except 1, which seems to be the recurring error.

```
======================================================================
ERROR: test_component_download_oscal_json (controls.tests.ComponentUITests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/codedata/code/govready-q-reuse/controls/tests.py", line 226, in test_component_download_oscal_json
    self.click_element('a[href="#oscal"]')
  File "/codedata/code/govready-q-reuse/siteapp/tests.py", line 149, in click_element
    elem = self.browser.find_element_by_css_selector(css_selector)
  File "/codedata/code/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 598, in find_element_by_css_selector
    return self.find_element(by=By.CSS_SELECTOR, value=css_selector)
  File "/codedata/code/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 976, in find_element
    return self.execute(Command.FIND_ELEMENT, {
  File "/codedata/code/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/codedata/code/venv/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":"a[href="#oscal"]"}
  (Session info: headless chrome=87.0.4280.88)
  (Driver info: chromedriver=2.38.552518 (183d19265345f54ce39cbb94cf81ba5f15905011),platform=Mac OS X 10.14.4 x86_64)
```

**Comment by gregelin** [comment]
Note that many of the changes are indentation changes. See commit history.

File: templates/components/element_detail_tabs.html [hunk 1]
```diff
@@ -292,7 +292,7 @@ <h4 id="panel-panel_num-title" class="panel-title">
                       </div>
                       <div class="form-group">
                         <input type="hidden" id="smt_id" name="smt_id" value="">
-                        <label for="statement">Statement</label>
+                        <label for="statement">What is the solution and how is it implemented?</label>
```
Comment (davidpofo) [inline templates/components/element_detail_tabs.html:295]: Not sure why this text is changed. Would be confusing for the user when the accordion tabs and items have statement in the text. To consistent with the label and how we currently use it in the backend reverting this makes sense.

File: templates/systems/element_detail_tabs.html [hunk 2]
```diff
@@ -87,176 +87,204 @@
...
+          <div id="above-tab-content" class="row">
+              <!-- Page Title-->
+              <div class="systems-title">
+                  <h2 class="control-heading">
+                      <a href="{{ project.get_absolute_url }}" class="systems-link">{{ system.root_element.name }}</a> &gt; <a href="/systems/{{system.id}}/components/selected" class="systems-link">Selected Components</a>
+                  </h2>
+              </div>
+              <!-- Control Lookup-->
+              <div class="systems-control-lookup">
+                  <!-- <form id="control-lookup" method="get" action="/controls/" onsubmit="show_control_by_id(); return false;">
```
Comment (davidpofo) [inline templates/systems/element_detail_tabs.html:102]: This section is also commented out in element_detail. Not sure if it is useful to keep.

File: templates/systems/element_detail_tabs.html [hunk 3]
```diff
@@ -87,176 +87,204 @@
...
-                    <label for="remarks">Remarks</label>
-                    <textarea class="" class="form-control systems-element-form-remarks" id="remarks_{{ forloop.counter }}" name="remarks" placeholder="Add remarks for team" cols="50">{{ smt.remarks }}</textarea>
+              </div>
+              <!-- Tab panel: combined -->
+              <!-- Combined control content does not appear on this page -->
+              <!-- Tab panel: component_controls -->
+              <div role="tabpanel" class="tab-pane" id="component_controls">
+                  <div id="control-description" class="control-text"><h3>Component Implementations Statements</h3></div>
+                  <div id="smt-list" class="systems-smt-list">
+                      <!-- Loop through existing component-control statements -->
```
Comment (davidpofo) [inline templates/systems/element_detail_tabs.html:155]: Most of this file could be formatted for easier reading

**Comment by woodt** [comment]
I checked out this branch, and it looks like it resolves the issue I ran into with correctly displaying the component statement editor.

I ran all the control tests against the branch, and there is one failure (as Greg notes above.)  There's also this in the test log:

```
2020-12-29 15:08:19,816 siteapp.good_settings_helpers level ERROR login ip=127.0.0.1 username=me result=user:2
2020-12-29 15:08:19,902 siteapp.views level INFO {"user": {"id": 2, "username": "me"}, "event": "project_list"}
2020-12-29 15:08:31,478 django.request level ERROR Internal Server Error: /systems/1/component/2
Traceback (most recent call last):
  File "/home/taw/.pyenv/versions/govready-q/lib/python3.8/site-packages/django/template/base.py", line 828, in _resolve_lookup
    current = current[bit]
TypeError: 'Statement' object is not subscriptable

During handling of the above exception, another exception occurred:
```

All the control tests pass in the develop branch, and that internal server error does not appear in the log.

File: templates/systems/element_detail_tabs.html (hunk 3 above)
Comment (gregelin) [reply templates/systems/element_detail_tabs.html:155]: Format is the way it is in order to make files more comparable with similar edit component control implementation statement pages.

File: templates/components/element_detail_tabs.html (hunk 1 above)
Comment (gregelin) [reply templates/components/element_detail_tabs.html:295]: Other pages actually read "What is the solution and how is it implemented?"

File: templates/systems/element_detail_tabs.html (hunk 2 above)
Comment (gregelin) [reply templates/systems/element_detail_tabs.html:102]: Controls search form has been commented out.

Comment (davidpofo) [reply templates/systems/element_detail_tabs.html:102]: I meant to say it should just be removed then if it is not used

File: templates/components/element_detail_tabs.html (hunk 1 above)
Comment (davidpofo) [reply templates/components/element_detail_tabs.html:295]: If I was a user this would confuse me.

**Pull request merged**